File: platform/src/main/java/ru/abondin/hreasy/platform/sec/SecurityUtils.java
package ru.abondin.hreasy.platform.sec;

import org.springframework.security.access.AccessDeniedException;
import reactor.core.publisher.Mono;
import ru.abondin.hreasy.platform.auth.AuthContext;


public class SecurityUtils {


    public static Mono<Boolean> validateUploadAvatar(AuthContext auth, int employeeId) {
        return Mono.defer(() -> {
            if (!auth.getAuthorities().contains("update_avatar") && employeeId != auth.getEmployeeInfo().getEmployeeId()) {
                return Mono.error(new AccessDeniedException("Only avatar owner or user with permission update_avatar can update the avatar"));
            }
            return Mono.just(true);
        });
    }

}

File: platform/src/main/kotlin/ru/abondin/hreasy/platform/sec/SecurityUtils.kt
package ru.abondin.hreasy.platform.sec

import org.springframework.security.access.AccessDeniedException
import ru.abondin.hreasy.platform.config.AuthContext


fun validateUploadAvatar(auth: AuthContext, employeeId: Int) {
    if (!auth.authorities.contains("update_avatar") && employeeId != auth.employeeInfo?.id) {
        throw AccessDeniedException("Only avatar owner or user with permission update_avatar can update the avatar");
    }
}

fun validateUpdateCurrentProject(auth: AuthContext, employeeId: Int) {
    if (!auth.authorities.contains("update_current_project_global") && employeeId != auth.employeeInfo?.id) {
        throw AccessDeniedException("Only logged in user or user with permission update_current_project_global can update the current project");
    }
}